sdk/search/search-documents/test/node/searchIndexerClient.spec.ts

```diff
@@ -0,0 +1,264 @@
+      await createDataSourceConnections(indexerClient);
+      await createSkillsets(indexerClient);
+      await createIndex(indexClient, TEST_INDEX_NAME);
+      await delay(5000);
+      await createIndexers(indexerClient);
+    }
+    recorder = record(this, environmentSetup);
+    // create the clients again, but hooked up to the recorder
+    ({ indexClient, indexerClient } = createClients<Hotel>(TEST_INDEX_NAME))
+  });
+
+
+  afterEach(async function() {
+    if (recorder) {
+      await recorder.stop();
+    }
+    if (!isPlaybackMode()) {
+      await indexClient.deleteIndex(TEST_INDEX_NAME);
+      await delay(2000);
+      await deleteDataSourceConnections(indexerClient);
+      await deleteSkillsets(indexerClient);
+      await deleteIndexers(indexerClient);
+    }
+  });
+
+  describe("#indexers", function() {
+    it("gets the list of indexers", async function(){
+      const indexers = await indexerClient.listIndexers();
+      assert.equal(indexers.length >= 5, true);
+    });
+
+    it("gets the list of indexer names", async function(){
+      const indexers = await indexerClient.listIndexersNames();
+      assert.equal(indexers.length >= 5, true);
+      assert.equal(indexers.includes("my-azure-indexer-1"), true);
+      assert.equal(indexers.includes("my-azure-indexer-2"), true);
+      assert.equal(indexers.includes("my-azure-indexer-3"), true);
+      assert.equal(indexers.includes("my-azure-indexer-4"), true);
+      assert.equal(indexers.includes("my-azure-indexer-5"), true);
+    });
+
+    it("gets the correct indexer object", async function(){
+      const indexer = await indexerClient.getIndexer("my-azure-indexer-1");
+      assert.equal(indexer.name, "my-azure-indexer-1");
+      assert.equal(indexer.dataSourceName, "my-data-source-5");
+      assert.equal(indexer.targetIndexName, "hotel-live-test2");
+      assert.equal(indexer.isDisabled, false);
+    });
+
+    it("throws error for invalid indexer object", async function(){
+      let retrievalError:boolean = false;
+      try {
+        await indexerClient.getIndexer("garbxyz");
+      } catch(ex) {
+        retrievalError = true;
+      }
+      assert.equal(retrievalError, true);
+    });
+
+    it("creates the indexer object using createOrUpdateIndexer", async function(){
+      let indexer: SearchIndexer = {
+        name: "my-azure-indexer-6",
+        description: "Description for Sample Indexer",
+        dataSourceName: "my-data-source-5",
+        targetIndexName: "hotel-live-test2",
+        isDisabled: false
+      };
+      await indexerClient.createOrUpdateIndexer(indexer);
+      indexer = await indexerClient.getIndexer("my-azure-indexer-6");
+      assert.equal(indexer.name, "my-azure-indexer-6");
+      assert.equal(indexer.dataSourceName, "my-data-source-5");
+      assert.equal(indexer.targetIndexName, "hotel-live-test2");
+      assert.equal(indexer.isDisabled, false);
+      await indexerClient.deleteIndexer(indexer);
```

Comment:
if any asserts fail, the indexer won't be deleted. Should this be in a `finally` block?

Reply:
I think it is safe to put in finally block. Done